stream/stream.go

```diff
@@ -28,10 +29,15 @@ func (e eventType) string() string {
 	return []string{"eventMatches", "eventError", "eventDone"}[e]
 }
 
+type Searcher interface {
+	zoekt.Searcher
+	StreamSearch(ctx context.Context, q query.Q, opts *zoekt.SearchOptions, sender Sender) (err error)
```

Comment:
what breaks if we add this to `zoekt.Searcher` rather than creating a new type?

edit: I am guessign we don't want things like indexSearcher to use a streaming interface? Although that might be alright? I suppose we "infect" too much of the codebase then?

edit2: I think `type Streamer` (currently Searcher here) and `type Sender` should probably move to the zoekt pkg. Then we keep the stream pkg for the client and handler?

Reply:
> I suppose we "infect" too much of the codebase then?

Yes, that was my primary intent. I wanted to keep as many changes as possible isolated to the stream package. `Searcher` is such a pervasive type that I don't think we should modify it in our fork.